src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/doobie/DoobiePetRepositoryInterpreter.scala

```diff
@@ -68,9 +74,12 @@ class DoobiePetRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
     extends PetRepositoryAlgebra[F] {
   import PetSQL._
 
-  def put(pet: Pet): F[Pet] =
+  def create(pet: Pet): F[Pet] =
     insert(pet).withUniqueGeneratedKeys[Long]("ID").map(id => pet.copy(id = id.some)).transact(xa)
 
+  override def update(pet: Pet): F[Option[Pet]] =
+    pet.id.traverse(PetSQL.update(pet, _).run.map(_ => pet)).transact(xa)
+
```

Comment:
This is rather dense.  Is there a newb way of writing this?

Reply:
Is it the usage of `traverse`? Or can we break up the expression somewhat, i.e.
 
```
  override def update(pet: Pet): F[Option[Pet]] = {
    def doUpdate(intoId: Long): ConnectionIO[Pet] = PetSQL.update(pet, intoId).run.map(_ => pet)
    pet.id.traverse(doUpdate).transact(xa)
  }
```

The part I dislike here this is the .`map(_ => pet)` part.